db/chart-migrations/2021-11-03-schema-cleanup.sql

```diff
@@ -0,0 +1,243 @@
+
+update charts
+set config = JSON_REMOVE(config, '$."margins"')
+where JSON_CONTAINS_PATH(config, 'one', '$."margins"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."add-country-control"')
+where JSON_CONTAINS_PATH(config, 'one', '$."add-country-control"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."logos"')
+where JSON_CONTAINS_PATH(config, 'one', '$."logos"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."units"')
+where JSON_CONTAINS_PATH(config, 'one', '$."units"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."map"."defaultProjection"')
+where JSON_CONTAINS_PATH(config, 'one', '$."map"."defaultProjection"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."map"."legendStepSize"')
+where JSON_CONTAINS_PATH(config, 'one', '$."map"."legendStepSize"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."identityLine"')
+where JSON_CONTAINS_PATH(config, 'one', '$."identityLine"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."isAutoSlug"')
+where JSON_CONTAINS_PATH(config, 'one', '$."isAutoSlug"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."onlyEntityMatch"')
+where JSON_CONTAINS_PATH(config, 'one', '$."onlyEntityMatch"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."isAutoTitle"')
+where JSON_CONTAINS_PATH(config, 'one', '$."isAutoTitle"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."map"."colorSchemeName"')
+where JSON_CONTAINS_PATH(config, 'one', '$."map"."colorSchemeName"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."map"."timeRanges"')
+where JSON_CONTAINS_PATH(config, 'one', '$."map"."timeRanges"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."highlightToggle"."object"')
+where JSON_CONTAINS_PATH(config, 'one', '$."highlightToggle"."object"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."map"."defaultYear"')
+where JSON_CONTAINS_PATH(config, 'one', '$."map"."defaultYear"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."chart-slug"')
+where JSON_CONTAINS_PATH(config, 'one', '$."chart-slug"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."map"."maxYear"')
+where JSON_CONTAINS_PATH(config, 'one', '$."map"."maxYear"') = 1;
+
+
+-- second batch
+
+update charts
+set config = JSON_REMOVE(config, '$."map"."timeInterval"')
+where JSON_CONTAINS_PATH(config, 'one', '$."map"."timeInterval"') = 1;
+
+update charts
+set config = JSON_REMOVE(config, '$."xAxis"."labelDistance"')
+where JSON_CONTAINS_PATH(config, 'one', '$."xAxis"."labelDistance"') = 1;
+
+
+update charts
+set config = JSON_SET(config, '$.map.targetYear', convert(config->>"$.map.targetYear", signed integer))
+where JSON_TYPE(JSON_EXTRACT(config, '$.map.targetYear')) = 'STRING' and
+JSON_EXTRACT(config, '$.map.targetYear') <> "latest"
+
+
+update charts
+set config = JSON_SET(config, '$.stackMode', null)
+where JSON_EXTRACT(config, '$.stackMode') = ''
```

Comment:
Could we delete it (undefined) instead of `null`? I think `null` is not handled in the codebase very well? At least TS doesn't expect `null` there.

There's actually more here that aren't handled in the codebase:

| stackMode | count |
|-|-|
| undefined | 1998 |
| null | 417 |
| "" | 3 |
| "absolute" | 1891 |
| "grouped" | 5 |
| "relative" | 83 |
| "stacked" | 2 |

I think the only valid values are `"relative"`, `"absolute"` and `undefined` (empty). Everything else can become `undefined`.